File: scripts/export-n8n-execution-summary.py
import sqlite3


DB = ".tmp/n8n-database.sqlite"
OUT = ".tmp/n8n-real-execution-summary.json"
RAW = ".tmp/n8n-execution-raw-flatted.json"
EXECUTION_ID = 17

conn = sqlite3.connect(DB)
cur = conn.cursor()
raw = cur.execute(
    'select data from execution_data where executionId = ?',
    (EXECUTION_ID,),
).fetchone()[0]

with open(RAW, "w", encoding="utf-8") as f:
    f.write(raw)

print(RAW)

File: scripts/export-n8n-execution-summary.js
const fs = require("fs");
const { parse } = require("flatted");

const raw = fs.readFileSync(".tmp/n8n-execution-raw-flatted.json", "utf8");
const data = parse(raw);
const runData = data.resultData.runData;
const summary = {};

for (const [nodeName, runs] of Object.entries(runData)) {
  if (!nodeName.startsWith("SoterAI")) continue;
  const main = runs[0]?.data?.main ?? [];
  const first = main[0]?.[0]?.json ?? {};
  summary[nodeName] = first;
}

fs.writeFileSync(".tmp/n8n-real-execution-summary.json", JSON.stringify(summary, null, 2));
console.log(".tmp/n8n-real-execution-summary.json");
